### PR TITLE
Issue #17 #18 #19 signup

### PR DESCRIPTION
**Pull request by julielee228**
#17 Check whether the account is already signed up
#18 Create database table, Inserting user information into the database
#19 Catch the error

### CODE REVIEW

File: Backend/router/v1/user.js [hunk 1]
```diff
@@ -13,15 +13,36 @@ router.all('/', (req, res) => {
...
   }).then(result => {
-    console.log("everything is good");
-    res.send(result);
+    if (result==""){
+
+      models.user.create({
+        email : req.body.email,
+        password : hashPassword,
+      }).then(result => {
+          console.log("< " + req.body.email + " > membership has been completed.");
```
Comment (mtnviewlab) [inline Backend/router/v1/user.js]: Do you know where this console.log it getting printed?
Like @shberry pointed, it is not a good idea to use console.log for production.

Comment (julielee228) [reply Backend/router/v1/user.js]: Sorry, it was my mistake. I deleted console but it seems to have been upload on github before the code was revised.

File: Backend/router/v1/user.js [hunk 2]
```diff
@@ -13,15 +13,34 @@ router.all('/', (req, res) => {
 });
 
 router.post('/signup', (req, res) => {
+
```
Comment (shberry) [inline Backend/router/v1/user.js]: remove extra line

File: Backend/router/v1/user.js [hunk 3]
```diff
@@ -13,15 +13,34 @@ router.all('/', (req, res) => {
...
-    password: hashPassword,
+  models.user.findAll({
+    where: {
+      email: req.body.email
+    }
+
   }).then(result => {
-    console.log("everything is good");
-    res.send(result);
+    if (result==""){
```
Comment (shberry) [inline Backend/router/v1/user.js]: usually it's easier to read if you add spaces around ==. "if (result == "")"

File: Backend/router/v1/user.js [hunk 4]
```diff
@@ -13,15 +13,34 @@ router.all('/', (req, res) => {
 });
 
 router.post('/signup', (req, res) => {
+
   let hashPassword = crypto.createHash("sha512").update(req.body.password).digest("hex");
```
Comment (shberry) [inline Backend/router/v1/user.js]: hashedPassword (hashPassword means you will hash the password, which is not this case)

File: Backend/router/v1/user.js [hunk 5]
```diff
@@ -13,15 +13,34 @@ router.all('/', (req, res) => {
...
-    console.log("everything is good");
-    res.send(result);
+    if (result==""){
+
+      models.user.create({
+        email : req.body.email,
+        password : hashPassword,
+      }).then(result => {
+          res.send(
+            "< " + req.body.email + " > membership has been completed."
```
Comment (shberry) [inline Backend/router/v1/user.js]: Is this the message that frontend will use to see if the new account was created or not?

File: Backend/router/v1/user.js (hunk 4 above)
Comment (julielee228) [reply Backend/router/v1/user.js]: hashedPassword -> encryption , Is it ok?

File: Backend/router/v1/user.js (hunk 2 above)
Comment (julielee228) [reply Backend/router/v1/user.js]: @shberry  I modified it.

File: Backend/router/v1/user.js (hunk 3 above)
Comment (julielee228) [reply Backend/router/v1/user.js]: @shberry I modified it.

File: Backend/router/v1/user.js (hunk 5 above)
Comment (julielee228) [reply Backend/router/v1/user.js]: It means membership registration completed.

Comment (shberry) [reply Backend/router/v1/user.js]: I know, I was asking if the frontend will be using this message to decided what to do. If that's the case, it's better to have these strings as constants in a separate file.

File: Backend/router/v1/user.js (hunk 4 above)
Comment (shberry) [reply Backend/router/v1/user.js]: I don't think encryption is a good variable name. It doesn't tell us what the variable holds. Why did you decided to make this change? hashedPassword seemed good go me (it reflects what the variable value means, right?).

File: Backend/router/v1/user.js (hunk 2 above)
Comment (shberry) [inline Backend/router/v1/user.js]: remove extra line?

File: Backend/router/v1/user.js [hunk 6]
```diff
@@ -13,15 +13,34 @@ router.all('/', (req, res) => {
...
+      });
+    } else {
+      res.send(
+        "< " + req.body.email + " > are already a member"
+      );
+    }
+  }).catch(err => {
+    res.send("err");
   });
+
```
Comment (shberry) [inline Backend/router/v1/user.js]: remove extra line

File: Backend/router/v1/user.js (hunk 4 above)
Comment (julielee228) [reply Backend/router/v1/user.js]: Oh, I didn't understand your comment. Now I understand it. I will change to hashedPassword. Thank you! :)

File: Backend/router/v1/user.js (hunk 2 above)
Comment (julielee228) [reply Backend/router/v1/user.js]: Yes!

File: Backend/router/v1/user.js (hunk 5 above)
Comment (julielee228) [reply Backend/router/v1/user.js]: @shberry  I don't know the separate file mean. Should I use real file or use json type?

Comment (shberry) [reply Backend/router/v1/user.js]: I'll explain more when we meet later. You can take care of it in another branch.

**Comment by shberry** [comment]
Approved

File: Backend/router/v1/user.js (hunk 5 above)
Comment (julielee228) [reply Backend/router/v1/user.js]: Yes!

**Pull request merged**